tools/pylib/_boutcore_build/boutcore.pyx.jinja

```diff
@@ -442,10 +478,27 @@ cdef class {{ field.field_type }}:
             if dims[i] != dims_in[i]:
                 raise IndexError("Expected %s but got %s."%(dims,dims_in))
 
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+{% for fun in "sqrt", "exp", "sin", "cos", "log", "abs" %}
+        if ufunc == np.{{ fun }}:
+            assert method == "__call__"
+            assert len(inputs) == 1
+            assert kwargs == {}
+            assert self == inputs[0]
+            return {{ field.fdd }}FromObj(c.{{ fun }}(self.cobj[0]))
+{% endfor %}
+        print("Debug output:", ufunc, method, inputs, kwargs)
+        if len(inputs) == 1:
+            assert self == inputs[0]
+            # convert to numpy and return
+            dat = self.get()
+            return dat.__array_ufunc__(ufunc, method, (dat,), **kwargs)
+        return NotImplemented
+
 {% for name, op in ops %}
     def __{{ name }}__(self,other):
-        fu = {{ field.field_type }}()
         import numbers
+        fu = {{ field.field_type }}()
```

Comment:
`fu` is not a great name. Is this really `result`?

Reply:
updated :+1: